crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -19,35 +22,45 @@ pub(super) struct InteractionPlugin;
 
 impl Plugin for InteractionPlugin {
     fn build(&self, app: &mut App) {
-        app.add_event::<MinimapClickEvent>().add_systems(
-            InputSchedule,
-            (
-                click_handler.in_set(InteractionSet::ClickHandler),
-                move_camera_system.after(InteractionSet::ClickHandler),
-                send_units_system
-                    .after(InteractionSet::ClickHandler)
-                    .before(CommandsSet::SendSelected),
-                delivery_location_system
-                    .after(InteractionSet::ClickHandler)
-                    .before(CommandsSet::DeliveryLocation),
-            )
-                .run_if(in_state(GameState::Playing)),
-        );
+        app.add_event::<MinimapPressEvent>()
+            .add_event::<MinimapDragEvent>()
+            .insert_resource(DraggingButtons(Vec::new()))
+            .add_systems(
+                InputSchedule,
+                (
+                    click_handler
+                        .in_set(InteractionSet::ClickHandler)
+                        .run_if(on_event::<MouseButtonInput>()),
+                    drag_handler
+                        .in_set(InteractionSet::DragHandler)
+                        .after(InteractionSet::ClickHandler)
+                        .run_if(on_event::<MouseMotion>()),
+                    move_camera_system.after(InteractionSet::DragHandler),
```

Comment:
Even though proper ordering is implied, I would be more explicitly here and not depend on other system dependencies: `.after(InteractionSet::ClickHandler).after(InteractionSet::DragHandler)`.